app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PostsController < ApplicationController
+  before_action :authenticate_member!, only: %i[new create]
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_member!, except: %i[index]
```

Comment:
- authenticate_member! is repeated twice and assigned for actions other that new and create. Kindly remove the last one.